zebra-consensus/src/block/check.rs

```diff
@@ -169,9 +169,25 @@ pub fn time_is_valid_at(
 ///
 /// `transaction_hashes` is a precomputed list of transaction hashes.
 pub fn merkle_root_validity(
+    network: Network,
     block: &Block,
     transaction_hashes: &[transaction::Hash],
 ) -> Result<(), BlockError> {
+    let nu = NetworkUpgrade::current(network, block.coinbase_height().expect("a valid height"));
```

Comment:
Nitpick: you might want to rename this to `block_nu`, to make the code a bit clearer.